commercial/app/model/merchandise/books/BookFinder.scala

```diff
@@ -2,67 +2,56 @@ package commercial.model.merchandise.books
 
 import akka.actor.ActorSystem
 import akka.pattern.CircuitBreaker
+import akka.util.Timeout
 import commercial.model.feeds.{FeedParseException, FeedReadException, FeedReader, FeedRequest}
-import common.Logging
+import commercial.model.merchandise.Book
+import common.{AkkaAgent, Logging}
 import conf.Configuration
 import conf.switches.Switches.BookLookupSwitch
-import commercial.model.merchandise.Book
 import play.api.libs.json._
 import play.api.libs.oauth.{ConsumerKey, OAuthCalculator, RequestToken}
 import play.api.libs.ws.{WSClient, WSSignatureCalculator}
-import shade.memcached.{Memcached, MemcachedCodecs, Configuration => MemcachedConfiguration}
 
 import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.control.NonFatal
 
 class BookFinder(actorSystem: ActorSystem, magentoService: MagentoService) extends Logging {
 
-  private implicit lazy val executionContext = actorSystem.dispatchers.lookup("akka.actor.memcached")
-
-  private lazy val defaultBookDataCache = new MemcachedBookDataCache(actorSystem)
-
-  def findByIsbn(isbn: String,
-                 cache: BookDataCache = defaultBookDataCache,
-                 lookup: String => Future[Option[JsValue]] = magentoService.findByIsbn):
-  Future[Option[Book]] = {
-
-    def cachedBook(bookData: JsValue): Future[Option[Book]] = {
-      Future.successful {
-        bookData match {
-          case JsNull => None
-          case _ =>
-            bookData.validate[Book] fold(
-              invalid => {
-                log.error(Json.stringify(JsError.toJson(invalid)))
-                None
-              },
-              book => Some(book))
-        }
+  private implicit val bookActorExecutionContext: ExecutionContext = actorSystem.dispatchers.lookup("akka.actor.book-lookup")
+  private implicit val bookActorTimeout: Timeout = 0.2.seconds
+  private implicit val magentoServiceImplicit = magentoService
+
+  def findByIsbn(isbn: String): Option[Book] = BookAgent.get(isbn) map { _.as[Book] }
+}
+
+object BookAgent extends Logging {
+
+  private lazy val cache = AkkaAgent(Map.empty[String, JsValue])
```

Comment:
In theory it can be as many as the number of (distinct) book reviews we have [on the site](http://content.guardianapis.com/search?tag=tone%2Freviews%2Cbooks%2Fbooks&api-key=test) (32,630). I think that this is manageable for the little `Map`, but perhaps it would be prudent to add a TTL for each book in the cache and do a bit of pruning when updating the agent. What do you think?